x-pack/plugins/ml/public/explorer/explorer_swimlane.js

```diff
@@ -118,81 +121,57 @@ export class ExplorerSwimlane extends React.Component {
     elements.map(e => d3.select(e).classed('ds-selected', false));
   }
 
-  cellClick(cellsToSelect, { laneLabels, bucketScore, times }) {
-    if (cellsToSelect.length > 1 || bucketScore > 0) {
-      this.selectCell(cellsToSelect, laneLabels, times, bucketScore, true);
-    } else {
-      this.clearSelection();
-    }
-  }
-
-  checkForSelection() {
-    const element = d3.select(this.rootNode.parentNode);
-
+  selectCell(cellsToSelect, { laneLabels, bucketScore, times }) {
     const {
-      appState,
+      selection,
+      mlExplorerDashboardService,
       swimlaneData,
       swimlaneType
     } = this.props;
 
-    // Check for selection in the AppState and reselect the corresponding swimlane cell
-    // if the time range and lane label are still in view.
-    const selectionState = appState.mlExplorerSwimlane;
-    const selectedType = _.get(selectionState, 'selectedType', undefined);
-    const viewBy = _.get(selectionState, 'viewBy', '');
-    if (swimlaneType !== selectedType && selectedType !== undefined) {
-      element.selectAll('.lane-label').classed('lane-label-masked', true);
-      element.selectAll('.sl-cell-inner').classed('sl-cell-inner-masked', true);
-    }
+    let triggerNewSelection = false;
 
-    if ((swimlaneType !== selectedType) ||
-      (swimlaneData.fieldName !== undefined && swimlaneData.fieldName !== viewBy)) {
-      // Not this swimlane which was selected.
-      return;
+    if (cellsToSelect.length > 1 || bucketScore > 0) {
+      triggerNewSelection = true;
     }
 
-    const cellsToSelect = [];
-    const selectedLanes = _.get(selectionState, 'selectedLanes', []);
-    const selectedTimes = _.get(selectionState, 'selectedTimes', []);
-    const selectedTimeExtent = d3.extent(selectedTimes);
+    // Check if the same cells were selected again, if so clear the selection,
+    // otherwise activate the new selection. The two objects are built for
+    // comparison because we cannot simply compare to "appState.mlExplorerSwimlane"
+    // since it also includes the "viewBy" attribute which might differ depending
+    // on whether the overall or viewby swimlane was selected.
+    const oldSelection = {
+      selectedType: selection.selectedType,
+      selectedLanes: selection.selectedLanes,
+      selectedTimes: selection.selectedTimes
+    };
```

Comment:
nit, looks like this could be done with the spread operator.
`const oldSelection = { ...selection };`

Reply:
`{ ...selection }` would include the `viewBy` attribute which we need to avoid having here for the comparison, so as discussed I'm leaving the code as is.